src/manifold/src/constructors.cpp

```diff
@@ -317,95 +317,127 @@ Manifold Manifold::Extrude(const CrossSection& crossSection, float height,
  * @param crossSection A set of non-overlapping polygons to revolve.
  * @param circularSegments Number of segments along its diameter. Default is
  * calculated by the static Defaults.
+ * @param revolveDegrees Number of degrees to revolve. Default is 360 degrees.
  */
 Manifold Manifold::Revolve(const CrossSection& crossSection,
-                           int circularSegments) {
-  auto polygons = crossSection.ToPolygons();
+                           int circularSegments,
+                           float revolveDegrees) {
+
+  Polygons polygons = crossSection.ToPolygons();
+
   if (polygons.size() == 0) {
     return Invalid();
   }
 
+  Rect bounds = crossSection.Bounds();
+
+  // Take the x>=0 slice.
+  if (bounds.min.x < 0) {
+    glm::vec2 min = bounds.min;
+    glm::vec2 max = bounds.max;
+    CrossSection posBoundingBox = CrossSection({{0.0, min.y},{max.x, min.y},
+                                                {max.x,max.y},{0.0,max.y}});
+
+    // Can't use RectClip as it has many failure cases.
```

Comment:
That seems odd - what failure cases have you seen?

Reply:
I wish I would have kept failing cases. I will try to recreate them.

Reply:
Even in the plain Revolve test case this doesn't seem to work.

```c++
  if (bounds.min.x < 0) {
    glm::vec2 min = bounds.min;
    glm::vec2 max = bounds.max;
    Rect boundingBox = Rect({0.0, min.y}, {max.x, max.y});
    // Can't use RectClip as it has many failure cases.
    polygons = crossSection.RectClip(boundingBox).ToPolygons();
  }

```

Result is not manifold-2, doesn't export. Am I constructing the Rect wrong?

Reply:
I'm not seeing a manifoldness failure, but I'm definitely reproducing a serious problem with `RectClip`: When I clip a polygon with a hole in half, I'm only getting the outer contour; intersection gives the correct result. @geoffder would you mind taking a look?

This is fine for this PR, thanks!